gnome-initial-setup/pages/account/gis-account-page.c

```diff
@@ -102,6 +102,32 @@ on_validation_changed (gpointer        page_area,
   update_page_validation (page);
 }
 
+
+static void
+hide_password_switch_if_needed (GisAccountPage *page)
+{
+  GKeyFile *conf_file = gis_driver_get_vendor_conf_file (GIS_PAGE (page)->driver);
+  GisAccountPagePrivate *priv = gis_account_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+
+  /* do nothing if the configuration file doesn't exist */
+  if (conf_file == NULL)
+    return;
+
+  /* check the conf file to see if the password switch should be hidden */
+  if (!g_key_file_get_boolean (conf_file, "page.account", "show-password-switch", &error) &&
+      error == NULL) {
+    gis_account_page_local_show_password_switch (GIS_ACCOUNT_PAGE_LOCAL (priv->page_local),
+                                                 FALSE);
+    return;
+  }
+
+  if (!g_error_matches (error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND) &&
+      !g_error_matches (error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_GROUP_NOT_FOUND))
+    g_warning ("Couldn't get the value for key 'show-password-switch' in group "
```

Comment:
Please make `show-password-switch` and `page.account` into `#define`s or `static const char *`s rather than open-coding them in this message and the lookup above.